src/txrebroadcast.cpp

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <chainparams.h>
+#include <consensus/consensus.h>
+#include <logging.h>
+#include <miner.h>
+#include <script/script.h>
+#include <txrebroadcast.h>
+#include <validation.h>
+
+/** We rebroadcast 3/4 of max block weight to reduce noise due to circumstances
+ *  such as miners mining priority transactions. */
+static constexpr unsigned int MAX_REBROADCAST_WEIGHT = 3 * MAX_BLOCK_WEIGHT / 4;
+
+/** Default minimum age for a transaction to be rebroadcast */
+static constexpr std::chrono::minutes REBROADCAST_MIN_TX_AGE = 30min;
+
+/** Maximum number of times we will rebroadcast a tranasaction */
+static constexpr int MAX_REBROADCAST_COUNT = 6;
+
+/** Minimum amount of time between returning the same transaction for
+ * rebroadcast */
+static constexpr std::chrono::hours MIN_REATTEMPT_INTERVAL = 4h;
+
+/** The maximum number of entries permitted in m_attempt_tracker */
+static constexpr int MAX_ENTRIES = 500;
+
+/** The maximum age of an entry ~3 months */
+static constexpr std::chrono::hours MAX_ENTRY_AGE = std::chrono::hours(3 * 30 * 24);
+
+std::vector<TxIds> TxRebroadcastHandler::GetRebroadcastTransactions()
+{
+    std::vector<TxIds> rebroadcast_txs;
+    std::chrono::microseconds start_time = GetTime<std::chrono::microseconds>();
+
+    // If there has not been a cache run since the last block, the fee rate
+    // condition will not filter out any transactions, so skip this run.
+    if (m_tip_at_cache_time == ::ChainActive().Tip()) return rebroadcast_txs;
+
+    BlockAssembler::Options options;
+    options.nBlockMaxWeight = MAX_REBROADCAST_WEIGHT;
+    options.m_skip_inclusion_until = start_time - REBROADCAST_MIN_TX_AGE;
+    options.check_block_validity = false;
+    options.blockMinFeeRate = m_cached_fee_rate;
+
+    // Use CreateNewBlock to identify rebroadcast candidates
+    auto block_template = BlockAssembler(m_mempool, Params(), options)
+                          .CreateNewBlock(m_chainman.ActiveChainstate(), CScript());
+    std::chrono::microseconds after_CNB_time = GetTime<std::chrono::microseconds>();
```

Comment:
Super small nit: Since this is new code - consider renaming as `after_cnb_time` to conform to the symbol naming conventions as described in the developer notes? :)